### PR TITLE
log about more memory than available and re-try in case of TotalMemory change

### PR DESCRIPTION
**Pull request by giggsoff**
Seems, we need additional information to solve periodic errors `Remaining memory bytes 0 app instance needs...` in tests.

Signed-off-by: Petr Fedchenkov <giggsoff@gmail.com>

### CODE REVIEW

File: pkg/pillar/cmd/zedmanager/memorysizemgmt.go [hunk 1]
```diff
@@ -5,8 +5,10 @@ package zedmanager
 
 import (
 	"fmt"
+	"strings"
 
 	"github.com/lf-edge/eve/pkg/pillar/types"
+	"github.com/sirupsen/logrus"
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedmanager/memorysizemgmt.go:11]: In zedmanager you don't need logrus; you can use log.XXX since log is set up before Run is called.

**Review by rvs — APPROVED** [review]
LGTM sans @eriknordmark comments

**Comment by giggsoff** [comment]
With this PR I can see `getRemainingMemory discrepancy: accounted apps: ; usedMemorySize: 786432000; deviceMemorySize: 672137216` in logs, which indicates, that for some reason we have fallback to Dom0 stats inside `hypervisor.xenContext.GetHostCPUMem`. Seems, it is resolvable with re-try of GetHostCPUMem in metric timer, so we can adjust our re-tries in zedmanager.

File: pkg/pillar/cmd/zedmanager/memorysizemgmt.go (hunk 1 above)
Comment (giggsoff) [reply pkg/pillar/cmd/zedmanager/memorysizemgmt.go]: Good point, thank you, fixed

File: pkg/pillar/cmd/zedmanager/zedmanager.go [hunk 2]
```diff
@@ -51,6 +51,7 @@ type zedmanagerContext struct {
 	GCInitialized        bool
 	checkFreedResources  bool // Set when app instance has !Activated
 	currentProfile       string
+	currentTotalMemoryMB uint64
```
Comment (eriknordmark) [inline pkg/pillar/cmd/zedmanager/zedmanager.go:54]: What is this for? You never set it.

**Review by eriknordmark — CHANGES_REQUESTED** [review]
I think it makes more sense to localize this in domainmgr/hypervisor by having the xen GetHostCPUMem return an error when it xen info fails or returns nothing, and having domainmgr not publish anything in that case.

The metrics task in domainmgr will retry so zedmanager will get an the data sooner or later.

**Comment by giggsoff** [comment]
> I think it makes more sense to localize this in domainmgr/hypervisor by having the xen GetHostCPUMem return an error when it xen info fails or returns nothing, and having domainmgr not publish anything in that case.
> 
> The metrics task in domainmgr will retry so zedmanager will get an the data sooner or later.

But in this case we will go into https://github.com/lf-edge/eve/blob/995412b8a8b5b5b16c21c1f76ecf5946e8a4ad84/pkg/pillar/cmd/zedmanager/updatestatus.go#L402-L410 without any repeat trigger again. Or I miss something?

**Comment by eriknordmark** [comment]
> without any repeat trigger again. Or I miss something?

Yeah you'd also need the retry, but it is easier and more clear to trigger that the first time types.HostMemory.TotalMemoryMB is set to a non-zero value by having xen return zero when it can't get/parse it.

File: pkg/pillar/cmd/zedmanager/zedmanager.go (hunk 2 above)
Comment (giggsoff) [reply pkg/pillar/cmd/zedmanager/zedmanager.go:54]: I use it in handleHostMemoryImpl

**Review by eriknordmark — APPROVED** [review]
LGTM

**Review by eriknordmark — APPROVED** [review]
LGTM - kick eden again

**Pull request merged**